frontend/src/pages/modelCatalog/screens/ModelCatalog.tsx

```diff
@@ -15,16 +15,35 @@ const ModelCatalog: React.FC = conditionalArea(
 )(() => {
   const { modelCatalogSources } = React.useContext(ModelCatalogContext);
   const renderStateProps = {
-    empty: modelCatalogSources.data.length === 0,
+    empty: modelCatalogSources.data.length === 0 || !modelCatalogSources.data[0]?.models?.length,
     emptyStatePage: (
       <EmptyModelCatalogState
         testid="empty-model-catalog-state"
-        title="Request access to model catalog"
-        description="To request access to model catalog, contact your administrator."
-        headerIcon={() => <img src={typedEmptyImage(ProjectObjectType.registeredModels)} alt="" />}
+        title={
+          modelCatalogSources.error
+            ? 'Unable to load model catalog'
+            : 'Request access to model catalog'
+        }
+        description={
+          modelCatalogSources.error
+            ? 'Refresh the page or try again later'
+            : 'To request access to model catalog, contact your administrator.'
+        }
```

Comment:
We don't need to check anything related to `modelCatalogSources.error` in the `emptyStatePage` -- if there's an error, the `loadError` prop below will be true and we won't be rendering this empty state at all. Instead, ApplicationsPage will use its internal error rendering which will use your `errorMessage` prop below.

We can put the whole EmptyModelCatalogState back the way it was here - the `loadError` and `errorMessage` props of ApplicationsPage are sufficient for rendering errors on these pages.